### PR TITLE
BlockId removal: refactor: BlockBackend::block|block_status

### PR DESCRIPTION
**Pull request by michalkucharczyk**
It changes the arguments of:
-  `BlockBackend::block`
-  `BlockBackend::block_status`

method from: `BlockId<Block>` to: `Block::Hash`

This PR is part of BlockId::Number refactoring analysis (paritytech/substrate#11292)

polkadot companion: paritytech/polkadot#6477
cumulus companion: paritytech/cumulus#2041

### CODE REVIEW

**Review by dmitry-markin — APPROVED** [review]

File: client/service/test/src/client/mod.rs [hunk 1]
```diff
@@ -1569,26 +1557,17 @@ fn returns_status_for_pruned_blocks() {
 		block_on(client.check_block(check_block_a1.clone())).unwrap(),
 		ImportResult::AlreadyInChain,
 	);
-	assert_eq!(
-		client.block_status(&BlockId::hash(check_block_a1.hash)).unwrap(),
-		BlockStatus::InChainPruned,
-	);
+	assert_eq!(client.block_status(check_block_a1.hash).unwrap(), BlockStatus::InChainPruned,);
```
Comment (bkchr) [inline client/service/test/src/client/mod.rs:1560]:
```suggestion
	assert_eq!(client.block_status(check_block_a1.hash).unwrap(), BlockStatus::InChainPruned);
```

File: client/service/test/src/client/mod.rs [hunk 2]
```diff
@@ -1569,26 +1557,17 @@ fn returns_status_for_pruned_blocks() {
...
 	assert_eq!(
 		block_on(client.check_block(check_block_a2.clone())).unwrap(),
 		ImportResult::AlreadyInChain,
 	);
-	assert_eq!(
-		client.block_status(&BlockId::hash(check_block_a2.hash)).unwrap(),
-		BlockStatus::InChainPruned,
-	);
+	assert_eq!(client.block_status(check_block_a2.hash).unwrap(), BlockStatus::InChainPruned,);
```
Comment (bkchr) [inline client/service/test/src/client/mod.rs:1565]:
```suggestion
	assert_eq!(client.block_status(check_block_a2.hash).unwrap(), BlockStatus::InChainPruned);
```

File: client/service/test/src/client/mod.rs [hunk 3]
```diff
@@ -1569,26 +1557,17 @@ fn returns_status_for_pruned_blocks() {
...
 	assert_eq!(
 		block_on(client.check_block(check_block_a3.clone())).unwrap(),
 		ImportResult::AlreadyInChain,
 	);
-	assert_eq!(
-		client.block_status(&BlockId::hash(check_block_a3.hash)).unwrap(),
-		BlockStatus::InChainWithState,
-	);
+	assert_eq!(client.block_status(check_block_a3.hash).unwrap(), BlockStatus::InChainWithState,);
```
Comment (bkchr) [inline client/service/test/src/client/mod.rs:1570]:
```suggestion
	assert_eq!(client.block_status(check_block_a3.hash).unwrap(), BlockStatus::InChainWithState);
```

File: primitives/consensus/common/src/block_validation.rs [hunk 4]
```diff
@@ -19,18 +19,18 @@
...
 use futures::FutureExt as _;
-use sp_runtime::{generic::BlockId, traits::Block};
+use sp_runtime::traits::Block;
 use std::{error::Error, future::Future, pin::Pin, sync::Arc};
 
 /// A type which provides access to chain information.
 pub trait Chain<B: Block> {
-	/// Retrieve the status of the block denoted by the given [`BlockId`].
-	fn block_status(&self, id: &BlockId<B>) -> Result<BlockStatus, Box<dyn Error + Send>>;
+	/// Retrieve the status of the block denoted by the given [`Block::hash`].
```
Comment (bkchr) [inline primitives/consensus/common/src/block_validation.rs:27]:
```suggestion
	/// Retrieve the status of the block denoted by the given [`Block::Hash`].
```

**Review by bkchr — APPROVED** [review]

**Review by melekes — APPROVED** [review]
👍 (code-wise; don't know enough to reason about business logic)

**Comment by michalkucharczyk** [comment]
bot rebase

**Comment by michalkucharczyk** [comment]
bot merge

**Comment by michalkucharczyk** [comment]
bot merge

**Pull request merged**